File: src/main/java/shateq/java/goku/Player.java
package shateq.java.goku;

import java.util.ArrayList;
import java.util.List;

public class Player {
    public static final Player PAUSE = new Player("PAUSE");

    private final String name;
    private int points;
    private Color lastColor;
    private final List<Player> opponents = new ArrayList<>();

    public Player(String name) {
        this.name = name;
        this.points = 0;
    }

    public void setColor(Color color) {
        this.lastColor = color;
    }

    public Color lastColor() {
        return this.lastColor;
    }

    public void addPoints(int num) {
        this.points += num;
    }

    public String name() {
        return name;
    }

    public int points() {
        return points;
    }
}

File: src/main/kotlin/shateq/goku/kotlin/Player.kt
package shateq.goku.kotlin

import shateq.goku.java.Color

data class Player(@get:JvmName("name") val name: String, @get:JvmName("index") val index: Int) {
    companion object {
        @JvmField val PAUSE = Player("PAUSE", 0)
    }

    val opponents: List<Player> = ArrayList()

    lateinit var lastColor: Color

    var wins: Int = 0
        @JvmName("wins") get
        private set

    var draws: Int = 0
        @JvmName("draws") get
        private set

    fun incWins() { this.wins += 1 }

    fun incDraws() { this.draws += 1 }
}